src/coap_client.c

```diff
@@ -486,6 +487,55 @@ enum golioth_status golioth_coap_client_observe_async(struct golioth_client *cli
     return GOLIOTH_OK;
 }
 
+enum golioth_status golioth_coap_client_observe_release_async(
+    struct golioth_client *client,
+    const char *path_prefix,
+    const char *path,
+    enum golioth_content_type content_type,
+    uint8_t *token,
+    size_t token_len,
+    void *arg)
+{
+    if (!client || !path)
+    {
+        return GOLIOTH_ERR_NULL;
+    }
+
+    if (!client->is_running)
+    {
+        GLTH_LOGW(TAG,
+                  "Client not running, dropping observe release request for path %s%s",
+                  path_prefix,
+                  path);
+        return GOLIOTH_ERR_INVALID_STATE;
+    }
+
+    golioth_coap_request_msg_t request_msg = {
+        .type = GOLIOTH_COAP_REQUEST_OBSERVE_RELEASE,
```

Comment:
That is part of the commit before introducing `GOLIOTH_COAP_REQUEST_OBSERVE_RELEASE`.